Pill 8 fails in both Darwin and NixOS

[8.1. Packaging GNU hello world](https://nixos.org/nixos/nix-pills/generic-builders.html#idm140737316150720)

I'm using 2.0.4 on Darwin and 2.0 on NixOS.

In Darwin (unsurprisingly):
```gcc
make[2]: ar: Command not found
make[2]: *** [Makefile:1550: lib/libhello.a] Error 127
make[2]: Leaving directory '/private/tmp/nix-build-hello.drv-0/hello-2.10'
make[1]: *** [Makefile:1878: install-recursive] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-hello.drv-0/hello-2.10'
make: *** [Makefile:2371: install] Error 2
builder for '/nix/store/2aynd426blaxjgk538zxrf5zylx1wy4f-hello.drv' failed with exit code 2
error: build of '/nix/store/2aynd426blaxjgk538zxrf5zylx1wy4f-hello.drv' failed
```
I am aware that `ar` does not get installed in Darwin by Nix.


In NixOS (surprisingly):
```gcc
these derivations will be built:
  /nix/store/wpn3c68vfbbr34qhg7fz9bsm2shibj4a-hello.drv
building path(s) ‘/nix/store/sbz6j3ksylj57yrf6asq12nnlgpkwxak-hello’
tar (child): gzip: Cannot exec: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
/nix/store/rcrxq7f78cvhkc9xv3a2l64nz5vnqbv4-hello_builder.sh: line 3: cd: hello-2.10: No such file or directory
/nix/store/rcrxq7f78cvhkc9xv3a2l64nz5vnqbv4-hello_builder.sh: line 4: ./configure: No such file or directory
make: *** No targets specified and no makefile found.  Stop.
make: *** No rule to make target 'install'.  Stop.
builder for ‘/nix/store/wpn3c68vfbbr34qhg7fz9bsm2shibj4a-hello.drv’ failed with exit code 2
error: build of ‘/nix/store/wpn3c68vfbbr34qhg7fz9bsm2shibj4a-hello.drv’ failed
```

Please noticed that I replaced `binutils-wrapped` with `binutils` in NixOS, as I get an `undefined variable` error otherwise.

And yes, I downloaded the sources to the current directory and added the correct names to the script.

--- a/pills/08/hello-builder.txt
+++ b/pills/08/hello-builder.txt
@@ -1,4 +1,4 @@
-export PATH="$gnutar/bin:$gcc/bin:$gnumake/bin:$coreutils/bin:$gawk/bin:$gzip/bin:$gnugrep/bin:$gnused/bin:$binutils_unwrapped/bin"
+export PATH="$gnutar/bin:$gcc/bin:$gnumake/bin:$coreutils/bin:$gawk/bin:$gzip/bin:$gnugrep/bin:$gnused/bin:$binutils/bin"
 tar -xzf $src
 cd hello-2.10
 ./configure --prefix=$out

--- a/pills/08/hello-nix.txt
+++ b/pills/08/hello-nix.txt
@@ -3,7 +3,8 @@ derivation {
   name = "hello";
   builder = "${bash}/bin/bash";
   args = [ ./hello_builder.sh ];
-  inherit gnutar gzip gnumake gcc binutils-unwrapped coreutils gawk gnused gnugrep;
+  inherit gnutar gzip gnumake gcc coreutils gawk gnused gnugrep;
+  binutils = binutils-unwrapped;
   src = ./hello-2.10.tar.gz;
   system = builtins.currentSystem;
 }

